lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -103,22 +110,37 @@ export const migrateGranuleRecord = async (
 
   let existingRecord;
 
-  try {
-    existingRecord = await granulePgModel.get(trx, {
-      granule_id: record.granuleId,
-      collection_cumulus_id: collectionCumulusId,
-    });
-  } catch (error) {
-    if (!(error instanceof RecordDoesNotExist)) {
-      throw error;
+  if (migrateAndOverwrite !== 'true') {
```

Comment:
This is the 'guts' of the change.   

If migrateAndOverWrite is not true, we want to: 

Get the existing record, and ignore RecordDoesNotExist from knex/postgres.
-  If we're migrating only files, throw RecordDoesNotExist if it occurred (as we cannot migrate files for a granule record that hasn't been migrated.   This shouldn't happen, however we don't know what state user dbs are actualy in.      If there *is* a record, return the cumulus_id and we don't have to migrate the granule
- Otherwise, perform all other 'is it already migrated tasks' as the original logic did. 

If migrateAndOverWrite *is* true, skip all 'does it exist' logic and proceed to update the granule and files.     It is *noteworthy* that because we're using the `@cumulus/db` `granule.upsert` method that all of the `createdAt`/granule status rules apply, particularly for running granules.

Reply:
Very minor: should we `toLowerCase` the `migrateAndOverwrite` and `migrateOnlyFiles` params? We have a lot of Python users that might toss in a `True` 😄

Reply:
Very minor: should we `toLowerCase` the `migrateAndOverwrite` and `migrateOnlyFiles` params? We have a lot of Python users that might toss in a `True` 😄

Reply:
that's fair, adding a validation method at the top of the call chain.    Open to needing it here too.

Reply:
That.... is not a horrible idea. 

Better yet: 1eeb416dca

Reply:
Nice! Love `normalizeBoolParam`